core/tee/tee_obj.c

```diff
@@ -70,10 +70,11 @@ void tee_obj_close(struct user_ta_ctx *utc, struct tee_obj *o)
 
 void tee_obj_close_all(struct user_ta_ctx *utc)
 {
-	struct tee_obj_head *objects = &utc->objects;
+	struct tee_obj *o;
+	struct tee_obj *next;
 
-	while (!TAILQ_EMPTY(objects))
-		tee_obj_close(utc, TAILQ_FIRST(objects));
```

Comment:
Hi @lackan,
I think you're right, the original code is correct. I suppose Coverity is confused by the use of `tqh_first` and `tqe_next`. This seems to be a common issue with Coverity (https://mail-index.netbsd.org/coverity-updates/2014/02/19/msg000045.html).

@vchong can we tell Coverity to ignore this kind of construct? Or at least ignore the particular occurrences?
I'll revert the bogus fixes.